### DISCUSSION TITLE
Logging metrics with compute()

### DISCUSSION THREAD

**Discussion by ss007k** [Lightning-AI/pytorch-lightning#6223] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Trying to understand compute() for logging. Could someone please tell if the self.log(train_acc..., on_epoch=True) is same as def training_epoch_end() which uses compute - code below

    def __init__():
        # ...
        self.train_acc = pl.metrics.Accuracy()
        self.valid_acc = pl.metrics.Accuracy()

    def training_step(self, batch, batch_idx):
        # ...
        self.train_acc(pred, target)
        self.log("train_acc", self.train_acc, on_step=False, on_epoch=True)
        self.log('train_loss', loss, on_step=True, on_epoch=False)

        return loss

    def training_epoch_end(self, training_step_outputs):
        self.log('train_acc_epoch', self.train_acc.compute())
        

    def validation_step(self, batch, batch_idx):
        # ...
        self.valid_acc(pred, target)
        self.log('val_loss', loss, prog_bar=True)
        self.log('val_acc', self.valid_acc, on_step=False, on_epoch=True, prog_bar=True)

        return loss
    
    
    def validation_epoch_end(self, validation_step_outputs):
        self.log('valid_acc_epoch', self.valid_acc.compute())

**Comment by SkafteNicki** [comment, marked as answer]
Yes, its the same. Here is the relevant code:
https://github.com/PyTorchLightning/pytorch-lightning/blob/4a8422c2dc870d29d26748e6aa01406d2e74fa48/pytorch_lightning/core/step_result.py#L325-L330
Setting `on_epoch=True` will essentially call `metric.compute` at the end of the epoch.

**Reply by ss007k** [reply]
Thank you for clarfiying
